ml/git-log-to-jsonl

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import git
+import json
+import shutil
+from pathlib import Path
+
+
+def decode_bytes(data: bytes) -> str:
+    """
+    Attempts to decode bytes using a list of common encodings.
+    """
+    encodings_to_try = [
+        "utf-8",
+        "latin-1",
+        "iso-8859-1",
+        "cp1252",
+        "ascii",
+    ]
+
+    for encoding in encodings_to_try:
+        try:
+            return data.decode(encoding)
+        except UnicodeDecodeError:
+            continue
+
+    # If all else fails, use a replacement handler to avoid crashing
+    return data.decode("utf-8", errors="replace")
+
+
+def get_git_log_as_jsonl(repo_path: str):
+    """
+    Opens a local git repository, converts its log to a JSONL file.
+    """
+    repo_dir = Path(repo_path)
+
+    # Check if the path is a valid git repository
+    if not (repo_dir / ".git").exists():
+        print(f"Error: '{repo_path}' is not a valid git repository.")
+        return False
+
+    print(f"Opening repository at {repo_path}...")
+    try:
+        repo = git.Repo(repo_path)
+    except git.exc.InvalidGitRepositoryError:
+        print(f"Error: Path '{repo_path}' does not contain a valid git repository.")
+        return False
+
+    print("Generating JSONL file...")
+
+    try:
+
+        for commit in repo.iter_commits("HEAD"):
+            commit_id = commit.hexsha
+            message = decode_bytes(commit.message.strip().encode())
+
+            # Handling diffs for the first commit vs. subsequent commits
+            if commit.parents:
+                diffs = commit.diff(commit.parents[0], create_patch=True)
+            else:
+                diffs = commit.diff(None, create_patch=True)
+
+            for diff in diffs:
+                patch_text = decode_bytes(diff.diff)
+                # Use a_path or b_path depending on if the file was added/deleted
+                file_path = diff.a_path if diff.a_path else diff.b_path
```

Comment:
I *think* you would want this to be 

```python
diff.b_path if diff.b_path else diff.a_path
```
so that you get the new path if it was renamed or added, else it'll fall back on the original path.